crates/bevy_ecs/src/reflect/world.rs

```diff
@@ -0,0 +1,245 @@
+use std::any::TypeId;
+
+use bevy_reflect::{Reflect, ReflectFromPtr, TraitTypeData, TraitTypeDataRelevance, TypeData};
+
+use crate::component::ComponentId;
+use crate::entity::Entity;
+use crate::prelude::{AppTypeRegistry, Mut, World};
+
+impl World {
+    /// Returns the [`TypeId`] of the underlying component type. Returns None if the component does not correspond to a Rust type.
+    pub fn component_type_id(&self, component_id: ComponentId) -> Option<TypeId> {
+        self.components()
+            .get_info(component_id)
+            .and_then(|n| n.type_id())
+    }
+
+    /// Retrieves an immutable `dyn T` reference to the given entity's Component of the given [`ComponentId`]
+    pub fn get_dyn_by_id<T: TraitTypeDataRelevance + ?Sized>(
```

Comment:
Is it possible to add the `?Sized` bound as a supertrait of `TraitTypeDataRelevance`? Or does that not work/isn't desirable?

Reply:
Doesn't work unfortunately. The implicit `Sized` bound comes from the type parameter `T`, not from the trait. All traits can have `?Sized` implementors, unless you explicitly add a `Sized` bound on the trait definition.